src/producer/browser/Full.js

```diff
@@ -29,30 +29,66 @@ const FullBrowserProducer = (context) => {
   const splitsUpdater = SplitChangesUpdater(context);
   const segmentsUpdater = MySegmentsUpdater(context);
 
+  let isSynchronizeSplitsRunning = false;
+
+  function synchronizeSplits() {
+    isSynchronizeSplitsRunning = true;
+    return splitsUpdater().finally(function() {
+      isSynchronizeSplitsRunning = false;
+    });
+  }
+
+  let isSynchronizeMySegmentsRunning = false;
+
+  function synchronizeMySegments(segmentList) {
+    isSynchronizeMySegmentsRunning = true;
+    return segmentsUpdater(undefined, segmentList).finally(function () {
```

Comment:
If you want no retries, use 0 instead of undefined. The default assignment of the variable might change, specially in a world where we modularize.